src/Repository.php

```diff
@@ -0,0 +1,316 @@
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return RunsApi
+     */
+    public function getRunsApi(): RunsApi
+    {
+        return $this->runsApi;
+    }
+
+    /**
+     * @return ProjectsApi
+     */
+    public function getProjectsApi(): ProjectsApi
+    {
+        return $this->projectsApi;
+    }
+
+    /**
+     * @return ResultsApi
+     */
+    public function getResultsApi(): ResultsApi
+    {
+        return $this->resultsApi;
+    }
+
+    public function init()
+    {
+        $this->client = new \GuzzleHttp\Client([
+            'headers' => [
+                'X-Platform' => sprintf('php=%s', phpversion()),
+                'X-Client' => sprintf('qase-phpunit=%s', Reporter::VERSION),
+            ],
+        ]);
+
+        $config = Configuration::getDefaultConfiguration()
+            ->setHost(getenv('QASE_API_BASE_URL'))
+            ->setApiKey('Token', getenv('QASE_API_TOKEN'));
+
+        $this->runsApi = new RunsApi($this->client, $config);
+        $this->projectsApi = new ProjectsApi($this->client, $config);
+        $this->resultsApi = new ResultsApi($this->client, $config);
+        $this->casesApi = new CasesApi($this->client, $config);
+        $this->suitesApi = new SuitesApi($this->client, $config);
+
+        $this->getCases();
+        $this->getSuites();
+    }
+
+    public function getCases(): void
+    {
+        $cases = $this->getAllEntities($this->casesApi->getCasesRequest($this->projectCode));
+
+        foreach ($cases as $case) {
+            $this->existingCases[$case['id']] = $case['title'];
+            $this->existingCasesGroupedBySuite[$case['suite_id'] ?: 0][$case['title']] = $case['id'];
+        }
+    }
+
+    public function getSuites(): void
+    {
+        $suites = $this->getAllEntities($this->suitesApi->getSuitesRequest($this->projectCode));
+
+        $suites = array_map(
+            static function (array $suite) {
+                return [
+                    'id' => $suite['id'],
+                    'parent_id' => $suite['parent_id'],
+                    'title' => $suite['title'],
+                ];
+            },
+            $suites
+        );
+
+        $suitesTree = $this->buildSuitesTree($suites);
+        $this->suitesFlat = $this->flatArray($suitesTree);
+    }
+
+    private function getAllEntities(Request $request): array
+    {
+        $entities = [];
+
+        $limit = 100;
+        $offset = 0;
+
+        do {
+            $rawResponse = $this->client->send(
+                $request,
+                [
+                    'query' => ['limit' => $limit, 'offset' => $offset,],
+                ]
+            );
+
+            $response = \GuzzleHttp\json_decode((string)$rawResponse->getBody(), true);
+
+            $entities = array_merge($entities, $response['result']['entities']);
+
+            $offset += $limit;
+        } while (count($response['result']['entities']) > 0 && count($entities) < $response['result']['total']);
```

Comment:
could you make it more readable, please?

```
$count  = count()
$total = ...

while ($count > 0 && $total > count())
```

also, I suggest make here cycle barrier to prevent infinite loop